server.go

```diff
@@ -814,6 +814,34 @@ func (s *Server) CopyMetastore(w io.Writer) error {
 	})
 }
 
+// CopyShard writes the requested shard to a writer.
+func (s *Server) CopyShard(w io.Writer, shardID uint64) error {
+	s.mu.RLock()
+
+	// Locate the shard.
+	sh, ok := s.shards[shardID]
+	if !ok {
+		s.mu.RUnlock()
+		return ErrShardNotFound
+	}
+	if sh.store == nil {
+		s.mu.RUnlock()
+		return ErrShardNotLocal
+	}
+
+	return sh.view(func(tx *shardtx) error {
+		s.mu.RUnlock() // Unlock so not held during long read.
```

Comment:
I believe this is the right thing to do. We need to hold the RLock until we start streaming the shard, but I certainly don't want to hold until the streaming completes.